src/predictor/gpu_predictor.cu

```diff
@@ -324,22 +336,31 @@ class GPUPredictor : public xgboost::Predictor {
                 h_nodes.begin() + h_tree_segments[tree_idx - tree_begin]);
     }
 
-    size_t i_batch = 0;
     size_t batch_offset = 0;
     for (auto &batch : dmat->GetRowBatches()) {
-      CHECK(i_batch == 0 || devices_.Size() == 1) << "External memory not supported for multi-GPU";
-      // out_preds have been sharded and resized in InitOutPredictions()
-      batch.offset.Shard(GPUDistribution::Overlap(devices_, 1));
+      GPUSet devices = GPUSet::All(param_.gpu_id, param_.n_gpus, batch.Size());
+      ConfigureShards(devices);
```

Comment:
For GPUPredictor, shards don't have any data-dependent state, only the trees and the device number. It would be clearer to remove `ConfigureShards()`, and filter out cases where a shard has 0 rows to predict.

Reply:
Done.